ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ContractsTable.scala

```diff
@@ -23,159 +22,63 @@ private[events] sealed abstract class ContractsTable extends PostCommitValidatio
-          case (ids, params) => (ids, BatchSql(insertContractQuery, params.head, params.tail: _*))
-        },
-        deletions = deletions,
-        transientContracts = transientContracts,
-      )
-  }
-
-  final class PreparedBatches(
-      val insertions: Option[(Set[ContractId], BatchSql)],
-      val deletions: Option[(Set[ContractId], BatchSql)],
-      val transientContracts: Set[ContractId],
-  )
-
-  def prepareBatchInsert(
-      ledgerEffectiveTime: Instant,
-      transaction: CommittedTransaction,
-      divulgedContracts: Iterable[DivulgedContract],
-  ): RawBatches = {
-
-    // Add the locally created contracts, ensuring that _transient_
-    // contracts are not inserted in the first place
-    val localContractStageChanges =
-      transaction
-        .fold(AccumulatingBatches(Map.empty, Map.empty, Set.empty)) {
-          case (batches, (_, node: Create)) =>
-            batches.insert(
-              contractId = node.coid,
-              insertion = new RawBatch.Contract(
-                contractId = node.coid,
-                templateId = node.coinst.template,
-                createArgument = node.coinst.arg,
-                createLedgerEffectiveTime = Some(ledgerEffectiveTime),
-                stakeholders = node.stakeholders,
-                key = node.key.map(k => Key.assertBuild(node.coinst.template, k.key.value)),
-              )
-            )
-          case (batches, (_, node: Exercise)) if node.consuming =>
-            batches.delete(
-              contractId = node.targetCoid,
-              deletion = deleteContract(node.targetCoid),
-            )
-          case (batches, _) =>
-            batches // ignore any event which is neither a create nor a consuming exercise
-        }
-
-    // Divulged contracts are inserted _after_ locally created contracts to make sure they are
-    // not skipped if consumed in this transaction due to the logic that prevents the insertion
-    // of transient contracts.
-    // We skip contracts that are either added or deleted as part of this transaction.
-    val divulgedContractsInsertions =
-      divulgedContracts.iterator.collect {
-        case contract if !localContractStageChanges.contains(contract.contractId) =>
-          contract.contractId -> new RawBatch.Contract(
-            contractId = contract.contractId,
-            templateId = contract.contractInst.template,
-            createArgument = contract.contractInst.arg,
-            createLedgerEffectiveTime = None,
-            stakeholders = Set.empty,
-            key = None,
-          )
-      }.toMap
-
-    localContractStageChanges
-      .copy(insertions = localContractStageChanges.insertions ++ divulgedContractsInsertions)
-      .prepare
-
+    val inserts = localInserts.toVector ++ divulgedInserts.toVector
+    ContractsTable.Executables(
+      deleteContracts = batch(deleteContractQuery, deletes),
+      insertContracts = batch(insertContractQuery, inserts)
```

Comment:
```suggestion
      insertContracts = batch(insertContractQuery, inserts),
```

Reply:
Addressed in https://github.com/digital-asset/daml/pull/8139